components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/RoleManagementServiceImpl.java

```diff
@@ -0,0 +1,208 @@
+    public List<RoleBasicInfo> getRoles(int limit, int offset, String sortBy, String sortOrder, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        return roleDAO.getRoles(limit, offset, sortBy, sortOrder, tenantDomain);
+    }
+
+    @Override
+    public List<RoleBasicInfo> getRoles(String filter, int limit, int offset, String sortBy, String sortOrder,
+            String tenantDomain) throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        return roleDAO.getRoles(filter, limit, offset, sortBy, sortOrder, tenantDomain);
+    }
+
+    @Override
+    public Role getRole(String roleID, String tenantDomain) throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        Role role = roleDAO.getRole(roleID, tenantDomain);
+        role.setPermissions(getPermissionListOfRole(roleID, tenantDomain));
+        return role;
+    }
+
+    @Override
+    public RoleBasicInfo updateRoleName(String roleID, String newRoleName, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        return roleDAO.updateRoleName(roleID, newRoleName, tenantDomain);
+    }
+
+    @Override
+    public void deleteRole(String roleID, String tenantDomain) throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        roleDAO.deleteRole(roleID, tenantDomain);
+    }
+
+    @Override
+    public List<UserBasicInfo> getUserListOfRole(String roleID, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        return roleDAO.getUserListOfRole(roleID, tenantDomain);
+    }
+
+    @Override
+    public RoleBasicInfo updateUserListOfRole(String roleID, List<String> newUserList, List<String> deletedUserList,
+            String tenantDomain) throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        roleDAO.addUsersToRole(roleID, newUserList, tenantDomain);
+        return roleDAO.removeUsersFromRole(roleID, deletedUserList, tenantDomain);
+    }
+
+    @Override
+    public List<GroupBasicInfo> getGroupListOfRole(String roleID, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        return roleDAO.getGroupListOfRole(roleID, tenantDomain);
+    }
+
+    @Override
+    public RoleBasicInfo updateGroupListOfRole(String roleID, List<String> newGroupNamesList,
+            List<String> deletedGroupNamesList, String tenantDomain) throws IdentityRoleManagementException {
+
+        RoleDAO roleDAO = RoleDAOImpl.getInstance();
+        roleDAO.addGroupsToRole(roleID, newGroupNamesList, tenantDomain);
+        return roleDAO.removeGroupsFromRole(roleID, newGroupNamesList, tenantDomain);
```

Comment:
same case I mentioned above

Reply:
Added a single method in the DAO layer to handle the `updateGroupListOfRole` as discussed.